### ISSUE TITLE
Git fetch substep fail if tag updated

### ISSUE THREAD

**Issue by YngveNPettersen** [buildbot/buildbot#4562]
I am presently seeing a number of build failures in our bot due to git fetch failing in a submodule.

The failure is due to an updated tag being refused (with message"(would clobber existing tag)").

In our setup this is happening with the Chromium V8 sub module (and (currently) its tag lkgr/7.2

I am not seeing this with the old build system, based on buildbot 0.8.4 (chromium variant).

What I am noticing is that the git fetch operation in the current version (1.7.0) is slightly different from 0.84. The newer version specifies the branch as e.g. "master", while the old version species the ref to fetch as "+master". The "+" makes fetch accept updates to the branches and tags, even when not fast forward.

Checking history, the "+branch" format was present in the worker Git class until it was retired after the master version of the class was added around 0.8.5 or 0.8.6, but that version did not transfer the "+branch" string

I would suggest that the "+branch" string is reintroduced.

**Comment by YngveNPettersen** [comment]
An update on this: I found that this problem is likely due to a change in Git 2.20.0:

> * "git push" into refs/tags/* hierarchy is rejected without getting
   forced, but "git fetch" (misguidedly) used the "fast forwarding"
   rule used for the refs/heads/* hierarchy; this has been corrected,
   which means some fetches of tags that did not fail with older
   version of Git will fail without "--force" with this version.
>  * The rules used by "git push" and "git fetch" to determine if a ref
   can or cannot be updated were inconsistent; specifically, fetching
   to update existing tags were allowed even though tags are supposed
   to be unmoving anchoring points.  "git fetch" was taught to forbid
   updates to existing tags without the "--force" option.

https://github.com/git/git/blob/master/Documentation/RelNotes/2.20.0.txt
